### DISCUSSION TITLE
Clarification on group_by setting in reduce transform

### DISCUSSION THREAD

**Discussion by jblang** [vectordotdev/vector#13310] [Q&A] [answered]
I wanted to ask about [this line](https://vector.dev/docs/reference/configuration/transforms/reduce/#group_by) from the docs:

> Each group is combined independently, allowing you to keep independent events separate. 

This is a bit confusing--does "each group" consist of all the fields in a single group_by, or is each field listed in the group_by section an independent group? And if the latter interpretation is correct, how would I accomplish grouping by a combination of fields? Could I combine the desired fields into an array and group by that, or would I need to concatenate them into a string?

**Comment by jszwedko** [comment, marked as answer]
Hi @jblang !

The `group_by` forms a "composite key" by which incoming events are grouped before the `reduce` is applied. So if you have `group_by = ["host", "region"]`, all incoming lines with `host = X and region = Y` will be grouped together.

**Reply by jblang** [reply]
OK, that's what I would have expected but then that line in the doc made me second guess.

**Reply by jszwedko** [reply]
I opened https://github.com/vectordotdev/vector/pull/13330 to update the docs. Does that read any clearer to you? I agree the original was a bit ambiguous.

**Reply by jblang** [reply]
Yes, that is clearer. I really appreciate how responsive you all are to feedback. Vector is a great project.
